instrumentation/metric/registry.go

```diff
@@ -79,14 +89,46 @@ func (r *inMemoryRegistry) String() string {
 	return s
 }
 
-func (r *inMemoryRegistry) ExportAll() map[string]interface{} {
+func (r *inMemoryRegistry) ExportAll() map[string]exportedMetric {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	all := make(map[string]interface{})
+	all := make(map[string]exportedMetric)
 	for _, m := range r.mu.metrics {
 		all[m.Name()] = m.Export()
 	}
 
 	return all
 }
+
+func (r *inMemoryRegistry) ExportAndRotate() map[string]exportedMetric {
+	results := r.ExportAll()
+
+	for _, m := range r.mu.metrics {
```

Comment:
what happens if we export all and then before we reset all, some more metrics get updated? do we merely lose some precision, or does it result in metric corruption?
maybe export and rotate should be an atomic operation per metric?